src/helpers/loadOptimize.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { Store } from '../app/types';
+import { assertWindow } from '../app/utils';
+import createTestStore from '../test/createTestStore';
+import loadOptimize from './loadOptimize';
+
+describe('loadOptimize', () => {
+  let store: Store;
+  let window: Window;
+
+  beforeEach(() => {
+    store = createTestStore();
+    window = assertWindow();
+    window.gtag = jest.fn();
+    window.dataLayer = [];
+
+    if (typeof document === 'undefined') {
+      throw new Error('JSDom not loaded');
+    }
+
+    const createEvent = document.createEvent.bind(document);
+    const originalCreateElement = document.createElement.bind(document);
+
+    document.createElement = jest.fn((type: string) => {
+      const element = originalCreateElement(type);
+      const event = createEvent('Event');
+      event.initEvent('load');
+      setTimeout(() => element.onload && element.onload(event), 1);
+
+      return element;
+    });
+  });
+
+  afterEach(() => {
+    if (typeof document === 'undefined') {
+      throw new Error('JSDom not loaded');
+    }
+
+    document.head.innerHTML = '';
+  });
+
+  it('injects correct <script> into head if in production', async() => {
+    const newLocation = {
+      ...window.location,
+      hostname: 'openstax.org',
+    };
+    delete (window as any).location;
+    window.location = newLocation;
+
+    await loadOptimize(window, store);
+
+    if (document && document.head) {
```

Comment:
if you would use `assertDocument` at the top then you don't have to check for this